### DISCUSSION TITLE
Question: how to mock orm:MikroOrm in a class that uses UseRequestContext decorator

### DISCUSSION THREAD

**Discussion by jearangoo** [mikro-orm/nestjs#49] [Q&A] [answered]
**Describe the bug**
hello i am trying to unit test a class that have the UseRequestContext decorator like this
```
@Controller()
export class AuthenticationController{
        // class need to have this.orm for the UseRequestContext decorator to work
	private readonly orm:MikroORM<MongoDriver>;
	private readonly authenticationService:AuthenticationService;
	
	constructor(
		orm:MikroORM<MongoDriver>,
		authenticationService:AuthenticationService
	){
		this.orm=orm;
		this.authenticationService=authenticationService;
		
	}
	
	@GrpcMethod()
	@UseRequestContext()
	login(data:LoginRequest.AsObject):Promise<LoginResponse.AsObject>{
		return this.authenticationService.login(data);
	}
	
}

``` 
but i can't mock the orm property of the class, this is the test file im using, i use MikroORM.init and use that value for the mock
```
describe('AuthenticationController',():void=>{
	let controller:AuthenticationController;

        beforeEach(async():Promise<void>=>{
		const module:TestingModule=await Test.createTestingModule({
			controllers:[AuthenticationController],
			providers:[
				AuthenticationService,
				{
					provide:MikroORM,
					useValue:await MikroORM.init<MongoDriver>({type:'mongo',entities:[User,UserRole],dbName:'test'})
				}
			]
		}).compile();
		
		controller=module.get<AuthenticationController>(AuthenticationController);
		authenticationService=module.get<AuthenticationService>(AuthenticationService);
		
	});
	
	it('should be defined',():void=>{
		expect(controller).toBeDefined();
	});

});

```
im getting this error because the MikroORM.init method is trying to connect to a database
```
Exceeded timeout of 5000 ms for a hook
```
i tried to set the option connect: false for the MikroORM.init method to no try to connect to any database like stated in this comment https://github.com/mikro-orm/nestjs-realworld-example-app/issues/7#issuecomment-633386892 
but typescript complains that the connect option is not a valid property
```
{
	provide:MikroORM,
        // @ts-ignore
	useValue:await MikroORM.init<MongoDriver>({type:'mongo',entities:[User,UserRole],dbName:'test',connect:false})
}
TS2345: Argument of type '{ type: "mongo"; entities: (typeof User | typeof UserRole)[]; dbName: string; connect: boolean; }' is not assignable to parameter of type 'Options<MongoDriver> | Configuration<MongoDriver>'.   Object literal may only specify known properties, and 'connect' does not exist in type 'Options<MongoDriver> | Configuration<MongoDriver>'.
```
i tried to ignore the typescript error with @ts-ignore but still i get the Exceeded timeout error

also if i pass a empty to mock
```
{
    provide:MikroORM,
    useValue:{}
}
```
i get this error
```
@UseRequestContext() decorator can only be applied to methods of classes that carry `orm: MikroORM`
``` 
if i pass a empty object to the MikroORM.init method
```
{
    provide:MikroORM,
    useValue:await MikroORM.init<MongoDriver>({})
}
```
i get an error too that says i must provide the options 
type:  'mongo', 'mysql', 'mariadb', 'postgresql' or 'sqlite' 
entities: non empty array of entitites
dbName or clientUrl

my question is: what is the recommended way to mock the orm:MikroORM property in a class that uses the UseRequestContext decorator?

**Versions**

| Dependency | Version |
| - | - |
| node | 16.13.0 |
| typescript | 4.5.4 |
| mikro-orm | 4.5.9 |
| your-driver | mongo |

**Comment by B4nan** [comment, marked as answer]
> but typescript complains that the connect option is not a valid property

Because it is not - it never was a config option, it is a second parameter to `MikroORM.init()`.

> if i pass a empty object to the MikroORM.init method

If you pass empty object to the init method, you obviously get validation errors for the required things you havent provided. 

```ts
useValue: await MikroORM.init<MongoDriver>({type:'mongo',entities:[User,UserRole],dbName:'test'}, false)
```

**Comment by jearangoo** [comment]
thank you very much @B4nan its now working passing false as second argument, i have not seen that
